eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -102,6 +102,11 @@ jobs:
           BuildSHA: $(Build.SourceVersion)
           RepoId: 'Azure/azure-sdk-for-java'
 
+      - script: |
+          python ./eng/scripts/inject_codesnippets.py
+        displayName: 'Replace @codesnippet with embedded pre element'
+        workingDirectory: $(Build.SourcesDirectory)
+        
```

Comment:
/CC @weshaggard @mitchdenny 
It's not clear to me if this is the correct place to be calling this or if this should be similar to what I'd done for the generate overview. This does differ from the replace relatively links as that is used across all the language and this is specific to Java. If it used the same approach as the generate overview stuff then it could be tested locally as part of the build with a flag instead of running a script across the entire repo and then building a given library (or submitting a PR and having to download artifacts). The code snippets for a given library are always going to be under that library's directory in the repo (typically under the /samples subdirectory) so processing as part of the library build would effectively work the same way. The changes would minimal and he could follow the same pattern which would be to declare a profile in [pom.client.xml](https://github.com/Azure/azure-sdk-for-java/blob/master/pom.client.xml#L1022) and then pass in that flag inside [archetype-sdk-client.yml](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L117). It would also mean that the python script would have to change to take a single directory as an argument.

Reply:
I'm planning on adjusting to act just like generate-overview after your feedback Jim.